options/posix/generic/pthread-stubs.cpp

```diff
@@ -354,11 +354,11 @@ namespace {
 	FutexLock key_mutex_;
 }
 
-int pthread_exit(void *ret_val) {
+[[noreturn]] void pthread_exit(void *ret_val) {
 	auto self = mlibc::get_current_tcb();
 
 	if (__atomic_load_n(&self->cancelBits, __ATOMIC_RELAXED) & tcbExitingBit)
-		return 0; // We are already exiting
+		goto exit;
```

Comment:
Should have mentioned in this discord, but we can avoid a goto here pretty easily. Make a static ``mlibc::do_exit`` function and call it from here and at the bottom of this function.

Reply:
Fixed, should be good to go now.